libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -96,6 +101,7 @@ + (instancetype)sharedInstanceForStore:(SFSmartStore *)store {
             syncMgr = [[self alloc] initWithStore:store];
             syncMgrList[key] = syncMgr;
         }
+        [[SalesforceSDKManager sharedManager] registerAppFeature:kSFAppFeatureSmartSync];
```

Comment:
Any app compiled with SmartSync will use that manager - instead you should move the registerAppFeature call to SFSmartSyncSyncManager.m